devel/namespaces/index.md

```diff
@@ -0,0 +1,167 @@
+
+* `index/` -
+   Artifact names with this prefix follow the index namespace, replacing the `.` separators with `/`.
+   For example, a task that indexes artifacts under `gaia.npm_cache.node-v4-2.linux-x64` would name its artifacts with the prefix `index/gaia/npm_cache/node-v4_2/linux-x64`.
+   Scopes for creating and reading such artifacts should follow the ownership hierarchy of the index routes.
+
+* `project/<project>/` -
+   Artifact names with this prefix are the responsibility of the project, which may have further namespace conventions.
+
+## Hooks
+
+Hooks are divided into "hook groups", for which the namespace is defined here.
+Within a hook group, the names are arbitrary (or defined by the project).
+
+* `project:<project>` - hooks for a speceific project
+
+## Worker Types
+
+Worker types are broken down into `<provisionerId>` and `<workerType>`.
+Within the AWS provisioner, worker types are currently not namespaced.
+All names are up for grabs -- [for now](https://bugzilla.mozilla.org/show_bug.cgi?id=1220686).
+
+## Worker IDs
+
+Worker IDs are broken down into `<workerGroup>` and `<workerId>`.
+In the present implementation, both of these are arbitrary strings.
+For workers started by the AWS provisioner, they are avaiability zone and instance ID, respectively.
+For other worker types, anything goes.
+
+## Caches
+
+Caches are not namespaced, but they [should be](https://bugzilla.mozilla.org/show_bug.cgi?id=1220684).
+
+## Secrets
+
+Secrets provide key-value storage governed by scopes.
+As such, it's very important that secrets not be unexpectedly made accessible to users who should not see them.
+Secret names have the following structure:
+
+* `project/<project>/` -
+  Secrets with this prefix are the exclusive domain of the given project.
+  Users not associated with a project should not be given scopes associated with the project's secrets!
+
+## Routes
+
+TODO
+
+## Indexes
+
+The index provides a nice, dot-separated hierarchy of names.
+
+* `buildbot` - the "old" index for Buildbot builds; do not use
+
+* `funsize.v1` -
+   Tasks indexed under this tree represent funsize tasks.
+   These are the responsibility of the release engineering team.
+
+* `gaia.npm_cache.<nodever>.<platform>.<revision>` -
+   Tasks indexed here have generated the `node_modules` directory required for the given revision.
+   These are the responsibility of teh B2G automation team.
+
+* `garbage` -
+   Anything goes under this index path.
+   Use this for development and experimentation.
+
+* `gecko.v1` - another "old" index for builds; do not use
+
+* `gecko.v2.<tree>.revision.<revision>.<platform>.<build>`,
+* `gecko.v2.<tree>.latest.<platform>.<build>` -
+   Index for Gecko build jobs, either by revision or for the latest job with the given platform and build.
+   These are the responsibility of the release engineering team.
```

Comment:
It's tempting to write here that say mshal is the principal owner of `gecko.v2.*`